src/main/java/ua/realestate/management/security/entity/SecurityUserEntity.java

```diff
@@ -0,0 +1,66 @@
+package ua.realestate.management.security.entity;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import ua.realestate.management.entity.UserEntity;
+import ua.realestate.management.entity.UserRole;
+
+import java.util.Collection;
+
+@AllArgsConstructor
+@Data
+public class SecurityUserEntity implements UserDetails {
```

Comment:
One more point for improvement is that I don't see why you have created your own implementation of UserDetails. Please, have a look at the already existing one from spring-security module **org/springframework/security/core/userdetails/User.java**